### PR TITLE
Feat : Added API integration

### PR DESCRIPTION
**Pull request by Omkar-Chaskar**
- Authentication (Login , Logout , SignUp)
- Note (Add , Edit , Remove)
- Archive (Add, Restore , Delete)
- Trash ( Restore Note , Delete , Add To Trash)

### CODE REVIEW

File: src/component/addnote.js [hunk 1]
```diff
@@ -1,88 +1,85 @@
...
-              label : state.label,
-              priority : state.priority,
-              date : new Date().toLocaleDateString()
-            }
-          });
-        }
-        }>Save</Link>
-      </div>
+        <section>
+        <ul className="note-priority-container list" style={{backgroundColor: state.color}}>
```
Comment (atharvabhanage02) [inline src/component/addnote.js:77]: Avoid using Inline styles

File: src/component/edit.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
+import "../styles.css";
+import { useNotes } from "../context/noteProvider";
+// import {Link} from "react-router-dom";
```
Comment (atharvabhanage02) [inline src/component/edit.js]: You can remove the commented code

File: src/component/header.js [hunk 3]
```diff
@@ -1,12 +1,36 @@
 import { Link } from "react-router-dom";
 import "../styles.css";
+import { useAuth } from "../context";
 
 export default function Header() {
+  const { logOutHandler, user } = useAuth();
```
Comment (atharvabhanage02) [inline src/component/header.js:6]: You can extract the loginStatus here
`const { logoutHandler , user: { loginStatus} }  = useAuth()`

File: src/component/toasters.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import { Toaster } from "react-hot-toast";
+
+export const Toasters = () => {
+  return (
+    <>
+    <Toaster
```
Comment (atharvabhanage02) [inline src/component/toasters.js:7]: Good use of Toasters

**Review by atharvabhanage02 — APPROVED** [review]
- All the mentioned functionalities are working fine
- UI looks great
- Code is clean and modular 
- Great use of Toaster for notifications
- Also good use of index.js files wherever required
- Suggested some minor changes else everything look fine and PR is good to merge

File: src/App.js [hunk 5]
```diff
@@ -4,22 +4,36 @@ import Header from "./component/header";
 import Navigation from "./component/navigation";
 import Home from "./component/home";
 import Archive from "./component/archive";
-import Label from "./component/label";
 import Trash from "./component/trash";
 import AddNote from "./component/addnote";
+import EditNote from "./component/edit";
+import Login from "./component/login";
+import Signup from "./component/signup";
```
Comment (sri-rishi) [inline src/App.js:11]: Suggestion: can import all components from index.js in components folder

File: src/component/addnote.js [hunk 6]
```diff
@@ -1,88 +1,85 @@
...
   return (
     <div className="AddNote">
-      <div className="add-note-container">
+      <form className="add-note-container" 
+      onSubmit={ (e) => {
+      e.preventDefault()
+      dispatch({ type: "SET_DATE", payload: dayjs().format("YYYY-MM-DD") })
+      addNewNoteHandler(state)
+      dispatch({ type: "CLEAR_NOTE" });
+      }}>
```
Comment (sri-rishi) [inline src/component/addnote.js:24]: Suggestion: can extract code from here in separate function and use that function here

File: src/component/addnote.js [hunk 7]
```diff
@@ -1,88 +1,85 @@
...
-              type : "SET_NOTE",
-              value : e.target.value,
-              key : "body"
-            })}/>
+          <textarea id="w3review" name="notebody" rows="6" cols="50" placeholder="Note body" className="note-textarea" value={state.content}
+                    onChange={(e) => dispatch({ type: "SET_CONTENT", payload: e.target.value })}/>
         </section>
+        <ul className="note-label-container list p1 bold">
+          {state.pin ?<li><MdPushPin name="pin item" className="p3 text-grey ptr" title="pinned note" onClick={() => {dispatch({ type: "SET_PIN", payload: !state.pin })}}/><p>Unpin Note</p></li> 
+          : <li><BsPin name="pin item" className="p3 text-grey ptr" title="unpined note" onClick={() => {dispatch({ type: "SET_PIN", payload: !state.pin })}}/><p>Pin Note</p></li> }
```
Comment (sri-rishi) [inline src/component/addnote.js]: Suggestion: can fix code indentation here so it get easy to read to other person

**Review by sri-rishi — APPROVED** [review]
functionality is working fine, just some suggestion which I think can be done else feel free to merge

**Comment by shivamSspirit** [comment]
some error throw when app land

**Comment by shivamSspirit** [comment]
extract handler features from htmlview to javascript section

**Pull request merged**